src/Module/BaseApi.php

```diff
@@ -160,6 +175,55 @@ public static function getRequest(array $defaults)
 		return $request;
 	}
 
+	/**
+	 * Set boundaries for the "link" header
+	 * @param array $boundaries
+	 * @param int $id
+	 * @return array
+	 */
+	protected static function setBoundaries(int $id)
+	{
+		if (!isset(self::$boundaries['min'])) {
+			self::$boundaries['min'] = $id;
+		}
+
+		if (!isset(self::$boundaries['max'])) {
+			self::$boundaries['max'] = $id;
+		}
+
+		self::$boundaries['min'] = min(self::$boundaries['min'], $id);
+		self::$boundaries['max'] = max(self::$boundaries['max'], $id);
+	}
+
+	/**
+	 * Set the "link" header with "next" and "prev" links
+	 * @return void
+	 */
+	protected static function setLinkHeader()
+	{
+		if (empty(self::$boundaries)) {
+			return;
+		}
+
+		$request = self::$request;
+
+		unset($request['min_id']);
+		unset($request['max_id']);
+		unset($request['since_id']);
+
+		$prev_request = $next_request = $request;
+
+		$prev_request['min_id'] = self::$boundaries['max'] + 1;
+		$next_request['max_id'] = self::$boundaries['min'] - 1;
```

Comment:
Mastodon is performing `id < max_id` so the + 1 is unwarranted. https://github.com/tootsuite/mastodon/blob/c3aef491d66aec743a3a53e934a494f653745b61/app/models/concerns/paginable.rb#L20

```suggestion
		$next_request['max_id'] = self::$boundaries['min'];
```